src/Microsoft.Security.Utilities.Core/CommonAnnotatedKey.cs

```diff
@@ -19,12 +19,19 @@ public static bool TryCreate(string key, out CommonAnnotatedKey secret)
             string base64EncodedSignature = key.Substring(76, 4);
             bool isDerived = key[DerivedKeyCharacterOffset] == 'D';
 
-            if (key.Length != 88 && key.Length != 84)
+            if (key.Length != IdentifiableSecrets.StandardCommonAnnotatedKeySize &&
+                key.Length != IdentifiableSecrets.LongFormCommonAnnotatedKeySize)
             {
                 return false;
             }
 
-            if (key.Length == 84)
+            bool longForm = key.Length == IdentifiableSecrets.LongFormCommonAnnotatedKeySize;
+
+            // This code path is intended to ensure that common annotated security keys are
+            // highly backwards compatible with the older identifiable keys format. This
+            // should only entail providing the missing compute checksum byte of the key if
+            // it is absent. The validation performed below is a fairly expensive operation.
```

Comment:
>expensive [](http://example.com/codeflow?start=72&length=9)

we actually might want a knob in the future to prevent the heavyweight validation.